data/pigui/modules/settings-window.lua

```diff
@@ -509,76 +507,62 @@ local optionsTabs = {
 	["controls"]=showControlsOptions
 }
 
-local optionModalOpen = false
-
-local function optionsWindow()
-	if ui.showOptionsWindow then
-		if not optionModalOpen then
-			ui.openPopup("Options")
```

Comment:
Thank you for cleaning up this style of code in particular! I'd much rather we call one function to open rather than have to check it every frame.

Reply:
Yeah, that was the part that was driving me up the wall with popups. It was very awkard to have to check both if the window should be rendered and if it was opened already.
It's a bit silly we have to keep track of the popup stack when ImGui is doing it too, but the code should be more manageable.